app/controllers/sessions_controller.rb

```diff
@@ -1,35 +1,47 @@
 class SessionsController < ApplicationController
   def new
+    if current_user
+      redirect_to home_path
+    end
   end
 
   def create
     if user = Admin.find_by_email(params[:email])
       if user.authenticate(params[:password])
         session[:user_id] = user.id
-        session[:is_admin] = true
+        session[:user] = "admin"
         redirect_to controller: 'home', action: 'index'
       else
-        flash[:notice] = "Failed to Log In! Password is invalid for #{params[:email]}"
+        flash[:notice] = "Failed to Log In! Password is invalid"
         redirect_to '/login'
       end
     elsif user = Student.find_by_email(params[:email])
       if user.authenticate(params[:password])
         session[:user_id] = user.id
-        session[:is_admin] = false
+        session[:user] = "student"
+        redirect_to controller: 'home', action: 'index'
+      else
+        flash[:notice] = "Failed to Log In! Password is invalid"
+        redirect_to '/login'
+      end
+    elsif user = Instructor.find_by_email(params[:email])
+      if user.authenticate(params[:password])
+        session[:user_id] = user.id
+        session[:user] = "instructor"
         redirect_to controller: 'home', action: 'index'
       else
-        flash[:notice] = "Failed to Log In! Password is invalid for #{params[:email]}"
+        flash[:notice] = "Failed to Log In! Password is invalid"
         redirect_to '/login'
       end
     else
-      flash[:notice] = "Failed to Log In! Account with email #{params[:email]} not found"
+      flash[:notice] = "Failed to Log In! Account not found"
```

Comment:
Don't include "Account not found", just do "Failed to Log In!". Do this for all above failed login messages.